src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala

```diff
@@ -0,0 +1,57 @@
+package com.mewna.mc.box.listeners
+
+import org.bukkit.Material
+import org.bukkit.event.player.PlayerInteractEvent
+import org.bukkit.event.{EventHandler, Listener}
+import org.bukkit.block.data.Ageable
+import org.bukkit.inventory.ItemStack
+import com.massivecraft.factions.{Board, FLocation, FPlayers}
+
+class CropHarvestListener extends Listener {
+  @EventHandler
+  def onPlayerInteract(event: PlayerInteractEvent): Unit = {
+    val block = event.getClickedBlock
+    if(block == null) {
+      return
+    }
+    val fPlayer = FPlayers.getInstance.getByPlayer(event.getPlayer)
+    val faction = fPlayer.getFaction
+    val location = block.getLocation
+    val factionAtBlock = Board.getInstance.getFactionAt(new FLocation(location))
+    if(faction.equals(factionAtBlock) || factionAtBlock.isWilderness) {
+      // If the faction is player's faction, or the faction is the wild, we can replant
+      val material = block.getType
+      if(isCrop(material)) {
+        val blockAsAgeable = block.getBlockData.asInstanceOf[Ageable]
+        if(blockAsAgeable.getAge == blockAsAgeable.getMaximumAge) {
+          // If the interacted block is a full crop, we harvest
+          val itemsToDrop = getApplicableDrop(material).filter(_.getAmount != 0)
```

Comment:
Remove the val assignment and just do the foreach straight on it.

Reply:
Readability

Reply:
Done